keps/sig-auth/4193-tokenrequest-requester-info-enhancements/README.md

```diff
@@ -0,0 +1,616 @@
+
+Future tokens will once again include this information/no adverse effects.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes (as noted above in the test plan)
+
+### Rollout, Upgrade and Rollback Planning
+
+Rolling this out will be done by enabling the feature flag on all control plane hosts.
+This should not have any issues/affect during upgrades.
+Rollback is done by removing/disabling the feature gate.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+If somehow the tokens issued during the time where the feature was enabled were to be rejected, these tokens would
+need to be re-requested.
+
+This isn't so much a failure of the rollback, rather it would be a bug in the actual feature (hence feature gating).
+
+During a skew where hosts have different enablement, users would get tokens with varying content if they made multiple
+requests. These tokens will all be usable against all apiservers though, so there won't be issues due to skew.
+
+###### What specific metrics should inform a rollback?
+
+* `authentication_attempts`
+* `authorization_attempts_total`
+* `serviceaccount_valid_tokens_total`
+
+New metrics that can be used to identify if the feature is in use:
+
+* `serviceaccount_authentication_pod_node_ref_verified_total`
+* `serviceaccount_authentication_bound_object_verified_total{bound_object_kind="Node"}`
+* `serviceaccount_bound_tokens_issued_pod_with_node_tokens_total`
+* `serviceaccount_bound_tokens_issued_total{bound_object_kind="Node"}`
+* `serviceaccount_bound_tokens_issued_with_identifier_total`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+New metrics:
+
+* `serviceaccount_authentication_pod_node_ref_verified_total` - new metric that is incremeneted when a token bound to a Pod has its Node reference verified
+* `serviceaccount_authentication_bound_object_verified_total{bound_object_kind="Node"}` - new metric that is incremeneted when a token bound to a Node has its reference verified
+* `serviceaccount_bound_tokens_issued_pod_with_node_tokens_total` - new metric that is incremented when a node ref is embedded into a bound Pod token (aka implicitly added)
+* `serviceaccount_bound_tokens_issued_total{bound_object_kind="Node"}` - new metric that is incremented whenever a bound token is issued that references a Node (explicitly added)
+* `serviceaccount_bound_tokens_issued_with_identifier_total` - new metric that is incremented whenever a token that contains an identifier/JTI is issued
```

Comment:
I'm fuzzy on how these would be used to determine if the feature was working properly (rather than to just check if the feature was enabled, which can be done with the feature enablement metric). I think @jeremyrickard asked a similar question in the PRR section.

I'm happy to defer specifics of the metrics to the implementation review, make sure the details here get updated with the result of that, and ensure this is finalized before beta.

Reply:
Similar.  We'd be looking for some kind of label indicating success or failure of verification perhaps?